nautobot/extras/models/metadata.py

```diff
@@ -135,3 +138,70 @@ def save(self, *args, **kwargs):
     def delete(self, *args, **kwargs):
         # TODO: should we behave like CustomFieldChoice and raise a ProtectedError if this choice is in use anywhere?
         super().delete(*args, **kwargs)
+
+
+class ObjectMetadata(BaseModel, ChangeLoggedModel):
+    metadata_type = models.ForeignKey(
+        to=MetadataType,
+        on_delete=models.CASCADE,
+        related_name="object_metadatas",
+    )
+    contact = models.ForeignKey(
+        to=Contact,
+        on_delete=models.SET_NULL,
+        related_name="object_metadatas",
+        null=True,
+    )
+    team = models.ForeignKey(
+        to=Team,
+        on_delete=models.SET_NULL,
+        related_name="object_metadatas",
+        null=True,
+    )
+    scoped_fields = JSONArrayField(
+        base_field=models.CharField(max_length=50),
```

Comment:
Maybe longer? We want to be able to include custom fields in the scoped fields and what's their max `key` length?